src/main/java/ortus/boxlang/web/bifs/HtmlClean.java

```diff
@@ -0,0 +1,117 @@
+		};
+	}
+
+	/**
+	 * This method cleans the provided HTML string using Jsoup.
+	 * The safeList argument determines the level of cleaning applied.
+	 * It returns a cleaned HTML string or an empty string if the input is null or empty.
+	 * <h2>Safe List Options</h2>
+	 * The valid values for safeList are:
+	 * <ul>
+	 * <li><code>basic</code>: Basic cleaning, removes all tags except for a few safe ones.</li>
+	 * <li><code>none</code>: No cleaning, returns the original HTML.</li>
+	 * <li><code>simpletext</code>: Removes all tags and returns plain text.</li>
+	 * <li><code>basicwithimages</code>: Basic cleaning but allows images.</li>
+	 * <li><code>relaxed</code>: More lenient cleaning, allows more tags.</li>
+	 * </ul>
+	 * <h2>Differences from ESAPI</h2>
+	 * This BIF uses Jsoup for HTML cleaning, which is different from ESAPI's approach.
+	 * Jsoup provides a more flexible and powerful way to clean HTML, allowing for more customization and control over the cleaning process.
+	 * It supports various safelists that can be used to define the level of cleaning applied to the HTML content.
+	 * <h2>Usage</h2>
+	 * This BIF can be used in BoxLang scripts to sanitize user input or any HTML
+	 * content that needs to be cleaned before rendering it in a web application.
+	 * It is particularly useful for preventing XSS (Cross-Site Scripting) attacks by
+	 * ensuring that only safe HTML content is rendered in the browser.
+	 * <h2>Example</h2>
+	 * 
+	 * <pre>
+	 * // Clean HTML content with basic safelist
+	 * var cleanedHtml = HtmlClean( html: "&lt;script&gt;alert('XSS')&lt;/script&gt;&lt;p&gt;Hello World!&lt;/p&gt;" );
+	 * // cleanedHtml will be "&lt;p&gt;Hello World!&lt;/p&gt;"
+	 * // Clean HTML content with relaxed safelist
+	 * var cleanedHtmlRelaxed = HtmlClean( html: "&lt;img src='image.jpg' /&gt;&lt;script&gt;alert('XSS')&lt;/script&gt;&lt;p&gt;Hello World!&lt;/p&gt;", safeList: "relaxed" );
+	 * // cleanedHtmlRelaxed will be "&lt;img src='image.jpg' /&gt;&lt;p&gt;Hello World!&lt;/p&gt;"
+	 * </pre>
+	 *
+	 * @param context   The context in which the BIF is being invoked.
+	 * @param arguments Argument scope for the BIF.
+	 *
+	 * @argument.html A HTML string to be cleaned.
+	 *
+	 * @argument.safeList The level of cleaning to apply, defaults to "basic".
+	 *
+	 * @return A cleaned HTML string or an empty string if the input is null or empty.
+	 */
+	public Object _invoke( IBoxContext context, ArgumentsScope arguments ) {
+		var	target			= arguments.getAsString( KeyDictionary.html );
+		var	safeListName	= resolveSafelist( arguments.getAsString( KeyDictionary.safeList ) );
+
+		if ( target == null || target.isEmpty() ) {
+			return "";
+		}
+
+		return Jsoup.clean( target, safeListName );
+	}
+
+	/**
+	 * Resolves the Safelist based on the provided name.
+	 *
+	 * @param name The name of the safelist to resolve.
+	 *
+	 * @return The resolved Safelist.
+	 */
+	private Safelist resolveSafelist( String name ) {
```

Comment:
The default behavior for [SafeList](https://www.javadoc.io/doc/org.jsoup/jsoup/latest/org/jsoup/safety/Safelist.html) is not to preserve relative links.  Should we make that the default for these safe lists?

> Configure this Safelist to preserve relative links in an element's URL attribute, or convert them to absolute links. By default, this is false: URLs will be made absolute (e.g. start with an allowed protocol, like e.g. http://.